operators/src/processing/raster_kernel/mod.rs

```diff
@@ -0,0 +1,737 @@
+                TileInformation {
+                    global_tile_position: [-1, 0].into(),
+                    tile_size_in_pixels: [3, 3].into(),
+                    global_geo_transform: TestDefault::test_default(),
+                },
+                GridOrEmpty::from(
+                    Grid2D::new([3, 3].into(), vec![18, 17, 16, 12, 11, 10, 6, 5, 4]).unwrap(),
+                ),
+            ),
+            RasterTile2D::new_with_tile_info(
+                TimeInterval::new_unchecked(10, 20),
+                TileInformation {
+                    global_tile_position: [-1, 1].into(),
+                    tile_size_in_pixels: [3, 3].into(),
+                    global_geo_transform: TestDefault::test_default(),
+                },
+                GridOrEmpty::from(
+                    Grid2D::new([3, 3].into(), vec![15, 14, 13, 9, 8, 7, 3, 2, 1]).unwrap(),
+                ),
+            ),
+        ];
+
+        MockRasterSource {
+            params: MockRasterSourceParams {
+                data: raster_tiles,
+                result_descriptor: RasterResultDescriptor {
+                    data_type: RasterDataType::I8,
+                    spatial_reference: SpatialReference::epsg_4326().into(),
+                    measurement: Measurement::Unitless,
+                    time: None,
+                    bbox: None,
+                    resolution: None,
+                },
+            },
+        }
+        .boxed()
+    }
+
+    #[tokio::test]
+    async fn test_ndvi_gaussian_filter() {
+        let mut exe_ctx = MockExecutionContext::test_default();
+        let ndvi_id = add_ndvi_dataset(&mut exe_ctx);
+
+        let operator = RasterKernel {
+            params: RasterKernelParams {
+                kernel: RasterKernelMethod::Convolution {
+                    matrix: vec![
+                        vec![1. / 16., 1. / 8., 1. / 16.],
+                        vec![1. / 8., 1. / 4., 1. / 8.],
+                        vec![1. / 16., 1. / 8., 1. / 16.],
+                    ],
+                },
+            },
+            sources: SingleRasterSource {
+                raster: GdalSource {
+                    params: GdalSourceParameters { data: ndvi_id },
+                }
+                .boxed(),
+            },
+        }
+        .boxed()
+        .initialize(&exe_ctx)
+        .await
+        .unwrap();
+
+        let processor = operator.query_processor().unwrap().get_u8().unwrap();
+
+        let query_rect = RasterQueryRectangle {
+            spatial_bounds: SpatialPartition2D::new((-180., 90.).into(), (180., -90.).into())
+                .unwrap(),
+            time_interval: TimeInstance::from(DateTime::new_utc(2014, 1, 1, 0, 0, 0)).into(),
+            spatial_resolution: SpatialResolution::one(),
+        };
+        let query_ctx = MockQueryContext::test_default();
+
+        // let result_stream = processor.query(query_rect, &query_ctx).await.unwrap();
```

Comment:
remove

Reply:
[fe84d11](https://github.com/geo-engine/geoengine/pull/644/commits/fe84d112615b6f6e4dd8a3bae110fa84a93a94f2) and [b6b3237](https://github.com/geo-engine/geoengine/pull/644/commits/b6b323705d0d990d2ce33577c0f17192939cefcb)